library/lib.ml

```diff
@@ -483,7 +483,22 @@ let add_section_replacement f g poly hyps =
     let sechyps,ctx = extract_hyps (vars,hyps) in
     let ctx = Univ.ContextSet.to_context ctx in
     let inst = Univ.UContext.instance ctx in
-    let subst, ctx = Univ.abstract_universes ctx in
+    (** FIXME: this should probably be done at an upper level, by storing the
+        name information in the section data structure. *)
+    let map lvl = match Univ.Level.name lvl with
+      | None -> (* Having Prop/Set/Var as section universes makes no sense *)
+        assert false
+      | Some na ->
+        try
+          let qid = Nametab.shortest_qualid_of_universe na in
+          Name (Libnames.qualid_basename qid)
+        with Not_found ->
+          (** Best-effort naming from the string representation of the level.
+              See univNames.ml for a similar hack. *)
+          Name (Id.of_string_soft (Univ.Level.to_string lvl))
+    in
+    let nas = Array.map_to_list map (Univ.Instance.to_array inst) in
```

Comment:
Maybe move this out as its own function `instance -> name list`. It would avoid the name hacking distracting from the logic of `add_section_replacement`.

Reply:
I was thinking about this, but I don't know where such function should belong. It is really a user-facing feature so it would naturally be somewhere up, but the current wart forces it to be placed quite low. For the time being it might be better to leave it here, I will fix this code soon enough.

Reply:
I was thinking just 
~~~ocaml
let name_instance = ...
let add_section_replacement = ...
~~~

Reply:
Ah, OK, indeed. Let's do that.